rmw_connext_cpp/src/functions.cpp

```diff
@@ -230,7 +230,7 @@ rmw_create_publisher(
     }
   }
 
-  if (!get_datawriter_qos(participant, qos_profile, datawriter_qos)) {
+  if (!get_datawriter_qos(participant, *qos_profile, datawriter_qos)) {
```

Comment:
It could I guess, but since it isn't pure C it doesn't have to. I was just going for minimal changes.